In [1]:
import pandas as pd
import numpy as np

## Question 1 Open the dataset as a pandas DataFrame named imdb

In [2]:
imdb = pd.read_csv("imdb_2019.tsv", sep='\t')
imdb.head()

index     tconst titleType                    primaryTitle  \
0   87501.0  tt0089435     short                           Kokoa   
1   89512.0  tt0091490     short             Martina's Playhouse   
2  114407.0  tt0116991     movie             Mariette in Ecstasy   
3  126556.0  tt0129960   tvMovie  Eine geschlossene Gesellschaft   
4  166388.0  tt0172112     short                        Ambulans   

                    originalTitle  isAdult  startYear  endYear  \
0                           Kokoa      0.0     2019.0      NaN   
1             Martina's Playhouse      0.0     2019.0      NaN   
2             Mariette in Ecstasy      0.0     2019.0      NaN   
3  Eine geschlossene Gesellschaft      0.0     2019.0      NaN   
4                        Ambulans      0.0     2019.0      NaN   

   runtimeMinutes  genres  
0            13.0     NaN  
1            20.0     NaN  
2             NaN     NaN  
3             NaN     NaN  
4            11.0     NaN

# Question 2 How many types of titles are there in the column titleType? No for loops allowed! Check pandas.unique, pandas.Series.value_counts, set

In [3]:
imdb[('titleType')].unique()

array(['short', 'movie', 'tvMovie', 'video', 'tvSeries', 'tvEpisode',
       'tvMiniSeries', 'tvSpecial', 'videoGame', 'tvShort'], dtype=object)

In [4]:
len(imdb[('titleType')].unique())

10

In [5]:
imdb.titleType.value_counts()

tvEpisode       225972
short            35351
movie            15640
video            10450
tvSeries          7154
tvMovie           2431
tvMiniSeries      2000
tvSpecial         1306
videoGame          884
tvShort            151
Name: titleType, dtype: int64

In [6]:
imdb[('titleType')].value_counts().to_frame()

titleType
tvEpisode        225972
short             35351
movie             15640
video             10450
tvSeries           7154
tvMovie            2431
tvMiniSeries       2000
tvSpecial          1306
videoGame           884
tvShort             151

### ANSWER : 10 different types of titles for column 'titleType'

## Question 3 Create a slice of imdb that only contains the following columns:  titleType, primaryTitle, startYear, runtimeMinutes

In [7]:
imdb_slice = imdb[['titleType','primaryTitle','startYear','runtimeMinutes']]
imdb_slice.tail()

titleType                                       primaryTitle  \
301334  tvEpisode  Talent Coaching with IMOR's Bianca Desmore Mit...   
301335  tvEpisode                                             Escape   
301336  tvEpisode                                      Tinne Oltmans   
301337  tvEpisode                                       Luc Janssens   
301338  tvEpisode                       Danira Boukhriss Terkessidis   

        startYear  runtimeMinutes  
301334     2019.0             NaN  
301335     2019.0             NaN  
301336     2019.0             NaN  
301337     2019.0             NaN  
301338     2019.0             NaN

## Question 4 Create a subset of imdb named tvEpisodes_2019 that only includes the type tvEpisodes

In [8]:
tvEpisodes_2019 = imdb[(imdb['titleType'] == 'tvEpisode')]
tvEpisodes_2019.head()

index      tconst  titleType        primaryTitle       originalTitle  \
21  758955.0   tt0782666  tvEpisode  Save the Duckling!  Save the Duckling!   
30  969426.0  tt10001058  tvEpisode       Le grand saut       Le grand saut   
32  969458.0  tt10001110  tvEpisode      The Interviews      The Interviews   
34  969464.0  tt10001120  tvEpisode         The New Guy         The New Guy   
35  969465.0  tt10001122  tvEpisode        The New Girl        The New Girl   

    isAdult  startYear  endYear  runtimeMinutes  genres  
21      0.0     2019.0      NaN             NaN     NaN  
30      0.0     2019.0      NaN            51.0     NaN  
32      0.0     2019.0      NaN             NaN     NaN  
34      0.0     2019.0      NaN             NaN     NaN  
35      0.0     2019.0      NaN             NaN     NaN

In [9]:
## checking if it worked. 
tvEpisodes_2019.titleType.value_counts()

tvEpisode    225972
Name: titleType, dtype: int64

## Question 5 Percentage of adult films over total releases in 2019. Check pd.Series.mean


In [10]:
imdb.isAdult.value_counts()

0.0    292394
1.0      8945
Name: isAdult, dtype: int64

In [11]:
imdb.isAdult.mean()

0.029684176293144928

### Answer: 2.9% of the movies are Adult Films. 

## Question 6 Create a column named words_in_title that contains the total number of words in the title. Use map alongside a function or a lambda function.


In [13]:
imdb["words_in_title"] = imdb["primaryTitle"].str.split().str.len()
## check if it worked. 
imdb.head()

index     tconst titleType                    primaryTitle  \
0   87501.0  tt0089435     short                           Kokoa   
1   89512.0  tt0091490     short             Martina's Playhouse   
2  114407.0  tt0116991     movie             Mariette in Ecstasy   
3  126556.0  tt0129960   tvMovie  Eine geschlossene Gesellschaft   
4  166388.0  tt0172112     short                        Ambulans   

                    originalTitle  isAdult  startYear  endYear  \
0                           Kokoa      0.0     2019.0      NaN   
1             Martina's Playhouse      0.0     2019.0      NaN   
2             Mariette in Ecstasy      0.0     2019.0      NaN   
3  Eine geschlossene Gesellschaft      0.0     2019.0      NaN   
4                        Ambulans      0.0     2019.0      NaN   

   runtimeMinutes  genres  words_in_title  
0            13.0     NaN             1.0  
1            20.0     NaN             2.0  
2             NaN     NaN             3.0  
3             NaN     NaN             3.0  
4            11.0     NaN             1.0

## Question 7 What’s the average value of runtimeMinutes for the type short?


In [37]:
short = imdb[(imdb['titleType'] == 'short')]
short.runtimeMinutes.mean()

12.536104279390065

In [47]:
imdb[(imdb['titleType'] == 'short')].runtimeMinutes.mean(skipna = True)

12.536104279390065

### Answer: Run time for short type is 12.54 minutes. 

## Question 8 Filter imdb to return tvMovie type with 3 or more words in the title, and less than 75 minutes of runTimeMinutes


In [18]:
imdb[(imdb.titleType == 'tvMovie') & (imdb.words_in_title >3) & (imdb.runtimeMinutes<75)].tail()

index     tconst titleType  \
300242  6586535.0  tt9906166   tvMovie   
300451  6587871.0  tt9908916   tvMovie   
301155  6591048.0  tt9915520   tvMovie   
301240  6591316.0  tt9916112   tvMovie   
301247  6591353.0  tt9916192   tvMovie   

                                          primaryTitle  \
300242  Flight MH370 5 Years On: The Mystery Continues   
300451           Restless Legs Syndrome: The New Cure?   
301155                          Fascism i en svart bil   
301240     A Priest, a Rabbi, and a Nun walk into a...   
301247      Danielle Darrieux: Il est poli d'être gai!   

                                         originalTitle  isAdult  startYear  \
300242  Flight MH370 5 Years On: The Mystery Continues      0.0     2019.0   
300451           Restless Legs Syndrome: The New Cure?      0.0     2019.0   
301155                          Fascism i en svart bil      0.0     2019.0   
301240     A Priest, a Rabbi, and a Nun walk into a...      0.0     2019.0   
301247      Danielle Darrieux: Il est poli d'être gai!      0.0     2019.0   

        endYear  runtimeMinutes  genres  words_in_title  
300242      NaN            56.0     NaN             8.0  
300451      NaN            60.0     NaN             6.0  
301155      NaN            55.0     NaN             5.0  
301240      NaN            21.0     NaN            10.0  
301247      NaN            53.0     NaN             7.0